include/PriorityQueue.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <iostream>
+#include <list>
+#include <exception>
+
+template <class T>
+struct MyComparator {
+	T operator()(T a, T b) const {
+		return a - b;
+	}
+};
+
+
+
+template <class T>
+class PriorityQueue { 
+public:
+	 
+	void push(const T& t); 
+	T poll();
+
+private:
+	std::list<T> m_queue;
+};
+
+template<class T>
+inline void PriorityQueue<T>::push(const T& t)
+{
+	auto iter = m_queue.begin();
+	while (iter != m_queue.end() && MyComparator<T>{}(*iter, t) < 0) {
+		++iter;
+	}
+	m_queue.insert(iter, t);
+}
+
+template<class T>
+inline T PriorityQueue<T>::poll()
+{
+		if (m_queue.empty())
+			throw std::runtime_error("PriorityQueue empty !");

```

Comment:
You should throw a more specific exception. It allows you to understand and handle specific scenarios.
Here is a list of standard exceptions: https://en.cppreference.com/w/cpp/error/exception